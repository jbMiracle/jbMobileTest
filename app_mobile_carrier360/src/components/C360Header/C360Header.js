import React, { PureComponent } from 'react';
import { Text, View, Image } from 'react-native';
import PropTypes from 'prop-types';
import { Header, Left, Right, Button, connectStyle } from 'native-base';
import { styleDef } from './C360Header.style';
import NotificationMenu from '../NotificationMenu/NotificationMenu';
import MoreMenu from '../MoreMenu/MoreMenu';
import C360Colors from '../../themes/360Colors';
import C360Icon from '../C360Icon/C360Icon';

const menuList = [
  { key: 'profile', icon: 'jbh-Contacts' },
  { key: 'manage_users', icon: 'jbh-User_Group' },
  { key: 'feedback', icon: 'jbh-Contact_Mail' },
  { key: 'sign_out', icon: 'jbh-Menu_Logout' },
];

const notificationsList = [
  { key: 'Load HV51606 - Status Update', status: 'You have been outbid', time: '2m' },
  {
    key: 'Favorite Lanes - Available Load',
    status: 'Dallas, TX, 73019 - San Antonio, TX, 78719',
    time: '23m',
  },
  { key: 'Manage Users - Approval Needed', status: 'New User Paul', time: '2d' },
  { key: 'Load HV51607 - Status Update', status: 'You have been outbid', time: '2m' },
  { key: 'Load HV51608 - Status Update', status: 'You have been served', time: '5m' },
];

const headerLogo = require('../../assets/images/360Logo-Reversed.png');
const demoProfileAvatar = require('../../assets/images/profileAvatar.jpeg');

class C360Header extends PureComponent {
  renderProfileAvatar = () => {
    // for now, avatarExists is always true for demo purposes, until we get actual accounts in the
    // app
    const avatarExists = true;
    const avatar = avatarExists
      ? demoProfileAvatar
      : demoProfileAvatar; // here, we will have the placeholder image

    return (
      <Image
        resizeMode='cover'
        style={ styleDef.profileAvatar }
        source={ avatar }
      />
    );
  };

  render() {
    const {
      showBackButton, backButtonOnPress, style, title,
    } = this.props;
    const { header } = this.props.style;
    return (
      <View>
        <Header style={ header } iosBarStyle='light-content' androidStatusBarColor={ C360Colors.Blue500 }>
          <Left>
            {renderBackButton({
              showBackButton, backButtonOnPress, style, title,
            })}
          </Left>
          <Right />

          <NotificationMenu notifications={ notificationsList } style={ style } />
          <MoreMenu menuList={ menuList } style={ style } />
        </Header>
      </View>
    );
  }
}

function renderBackButton({
  showBackButton, backButtonOnPress, style, title,
}) {
  const { backButtonStyle, headerLogoStyle } = style;

  if (showBackButton) {
    return (
      <Button iconLeft transparent light testID='header_backButton' onPress={ backButtonOnPress }>
        <C360Icon name='jbh-Chevron_Back' color='white' size={ 20 } />
        <Text style={ backButtonStyle }>{title}</Text>
      </Button>
    );
  }

  return <Image style={ headerLogoStyle } source={ headerLogo } />;
}

export default connectStyle('C360.Header', styleDef)(C360Header);

C360Header.propTypes = {
  showBackButton: PropTypes.bool.isRequired,
  backButtonOnPress: PropTypes.func.isRequired,
  style: PropTypes.object.isRequired,
  title: PropTypes.string.isRequired,
};
