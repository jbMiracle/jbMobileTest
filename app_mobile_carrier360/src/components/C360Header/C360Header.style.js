import { TouchableHighlight, Dimensions } from 'react-native';
import C360Colors from '../../themes/360Colors';

const popUpWidth = Dimensions.get('window').width;

export const styleDef = {
  header: {
    backgroundColor: C360Colors.Blue500,
    borderBottomWidth: 0,
    elevation: 0,
    alignItems: 'center',
  },
  backButtonStyle: {
    color: 'white',
    fontWeight: 'bold',
    paddingLeft: 10,
  },
  headerLogoStyle: {
    marginLeft: 10,
  },
  profileAvatar: {
    width: 40,
    height: 40,
    borderRadius: 40 / 2,
    marginRight: 10,
  },
  headerIcon: { justifyContent: 'center' },
  profileMenuView: {
    borderBottomWidth: 1,
    borderBottomColor: C360Colors.Grey500,
    flexDirection: 'row',
    justifyContent: 'flex-start',
    alignItems: 'center',
    paddingBottom: 12,
    marginBottom: 2,
  },
  mainText: { color: C360Colors.Grey800 },
  subText: {
    color: C360Colors.Grey600,
    fontSize: 11,
  },
  notificationRow: {
    flexDirection: 'row',
    flex: 1,
    paddingHorizontal: 2,
    paddingVertical: 2,
  },
  notificationTextView: {
    flexDirection: 'column',
    paddingHorizontal: 12,
  },
  notificationHierarchy: {
    fontWeight: 'bold',
    color: C360Colors.Grey800,
  },
  moreNotificationTextStyle: {
    color: C360Colors.Blue500,
    fontWeight: 'bold',
    fontSize: 13,
  },
  moreNotificationButtonStyle: {
    padding: 15,
    paddingTop: 25,
  },
  itemSeparatorStyle: {
    backgroundColor: C360Colors.Grey500,
    height: 1,
    opacity: 0.4,
  },
  timeStamp: {
    flexDirection: 'row',
    justifyContent: 'flex-start',
    width: 40,
  },
  badgeStyle: {
    backgroundColor: C360Colors.Yellow500,
    width: 10,
    height: 10,
  },
  flexColumn: { flexDirection: 'column' },
  alignFlexEnd: { alignItems: 'flex-end' },
  whiteText: { color: 'white' },
  alignCenter: { alignItems: 'center' },
  fontSize10: { fontSize: 10 },
  marginRight25: { marginRight: 25 },
  menuTriggerStyles: {
    triggerOuterWrapper: {
      marginRight: 10,
    },
    triggerText: { color: 'white' },
    triggerWrapper: {
      backgroundColor: 'transparent',
      alignItems: 'center',
      justifyContent: 'center',
    },
    triggerTouchable: {
      underlayColor: 'transparent',
      activeOpacity: 0.6,
    },
    TriggerTouchableComponent: TouchableHighlight,
  },
  popoverProps: {
    preferredPlacement: 'bottom',
    placement: 'bottom',
  },
};

export const menuOptionsStyle = ({ type }) => ({
  optionsContainer: {
    backgroundColor: C360Colors.White,
    alignItems: 'center',
    borderRadius: 7,
    // marginTop: 12.5,
  },
  optionsWrapper: {
    backgroundColor: C360Colors.White,
    borderRadius: 7,
    width: type === 'notifications' ? (popUpWidth / 1.033) : (popUpWidth / 1.67),
    paddingHorizontal: type === 'notifications' ? 3 : 20,
    paddingVertical: 10,
  },
  optionWrapper: {
    backgroundColor: 'transparent',
    flexDirection: 'row',
    alignItems: 'center',
    margin: 5,
  },
  optionTouchable: {
    underlayColor: 'transparent',
    activeOpacity: 0.3,
  },
  optionText: {
    color: C360Colors.Grey800,
    fontSize: 12,
  },
});
