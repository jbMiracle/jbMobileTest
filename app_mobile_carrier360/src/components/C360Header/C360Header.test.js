import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import C360Header from './C360Header';

it('can render an empty component', () => {
  const renderer = new ShallowRenderer();

  const props = {
    showBackButton: false,
    backButtonOnPress: () => {},
    title: 'Back',
  };

  renderer.render(<C360Header { ...props } />);

  expect(renderer.getRenderOutput()).toMatchSnapshot();
});

it('renders a header with the back button enabled', () => {
  const renderer = new ShallowRenderer();

  const props = {
    showBackButton: true,
    backButtonOnPress: () => {},
    title: 'Load Details',
  };

  renderer.render(<C360Header { ...props } />);

  expect(renderer.getRenderOutput()).toMatchSnapshot();
});
