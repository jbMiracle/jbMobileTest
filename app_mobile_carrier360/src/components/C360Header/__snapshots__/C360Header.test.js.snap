// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can render an empty component 1`] = `
<C360Header
  backButtonOnPress={[Function]}
  showBackButton={false}
  style={
    Object {
      "alignCenter": Object {
        "alignItems": "center",
      },
      "alignFlexEnd": Object {
        "alignItems": "flex-end",
      },
      "backButtonStyle": Object {
        "color": "white",
        "fontWeight": "bold",
        "paddingLeft": 10,
      },
      "badgeStyle": Object {
        "backgroundColor": "#FFDB00",
        "height": 10,
        "width": 10,
      },
      "flexColumn": Object {
        "flexDirection": "column",
      },
      "fontSize10": Object {
        "fontSize": 10,
      },
      "header": Object {
        "alignItems": "center",
        "backgroundColor": "#007DBA",
        "borderBottomWidth": 0,
        "elevation": 0,
      },
      "headerIcon": Object {
        "justifyContent": "center",
      },
      "headerLogoStyle": Object {
        "marginLeft": 10,
      },
      "itemSeparatorStyle": Object {
        "backgroundColor": "#C9C9C9",
        "height": 1,
        "opacity": 0.4,
      },
      "mainText": Object {
        "color": "#444444",
      },
      "marginRight25": Object {
        "marginRight": 25,
      },
      "menuTriggerStyles": Object {
        "TriggerTouchableComponent": [Function],
        "triggerOuterWrapper": Object {
          "marginRight": 10,
        },
        "triggerText": Object {
          "color": "white",
        },
        "triggerTouchable": Object {
          "activeOpacity": 0.6,
          "underlayColor": "transparent",
        },
        "triggerWrapper": Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "justifyContent": "center",
        },
      },
      "moreNotificationButtonStyle": Object {
        "paddingBottom": 15,
        "paddingLeft": 15,
        "paddingRight": 15,
        "paddingTop": 25,
      },
      "moreNotificationTextStyle": Object {
        "color": "#007DBA",
        "fontSize": 13,
        "fontWeight": "bold",
      },
      "notificationHierarchy": Object {
        "color": "#444444",
        "fontWeight": "bold",
      },
      "notificationRow": Object {
        "flex": 1,
        "flexDirection": "row",
        "paddingBottom": 2,
        "paddingLeft": 2,
        "paddingRight": 2,
        "paddingTop": 2,
      },
      "notificationTextView": Object {
        "flexDirection": "column",
        "paddingLeft": 12,
        "paddingRight": 12,
      },
      "popoverProps": Object {
        "placement": "bottom",
        "preferredPlacement": "bottom",
      },
      "profileAvatar": Object {
        "borderRadius": 20,
        "height": 40,
        "marginRight": 10,
        "width": 40,
      },
      "profileMenuView": Object {
        "alignItems": "center",
        "borderBottomColor": "#C9C9C9",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "marginBottom": 2,
        "paddingBottom": 12,
      },
      "subText": Object {
        "color": "#939393",
        "fontSize": 11,
      },
      "timeStamp": Object {
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "width": 40,
      },
      "whiteText": Object {
        "color": "white",
      },
    }
  }
  title="Back"
  virtual={undefined}
/>
`;

exports[`renders a header with the back button enabled 1`] = `
<C360Header
  backButtonOnPress={[Function]}
  showBackButton={true}
  style={
    Object {
      "alignCenter": Object {
        "alignItems": "center",
      },
      "alignFlexEnd": Object {
        "alignItems": "flex-end",
      },
      "backButtonStyle": Object {
        "color": "white",
        "fontWeight": "bold",
        "paddingLeft": 10,
      },
      "badgeStyle": Object {
        "backgroundColor": "#FFDB00",
        "height": 10,
        "width": 10,
      },
      "flexColumn": Object {
        "flexDirection": "column",
      },
      "fontSize10": Object {
        "fontSize": 10,
      },
      "header": Object {
        "alignItems": "center",
        "backgroundColor": "#007DBA",
        "borderBottomWidth": 0,
        "elevation": 0,
      },
      "headerIcon": Object {
        "justifyContent": "center",
      },
      "headerLogoStyle": Object {
        "marginLeft": 10,
      },
      "itemSeparatorStyle": Object {
        "backgroundColor": "#C9C9C9",
        "height": 1,
        "opacity": 0.4,
      },
      "mainText": Object {
        "color": "#444444",
      },
      "marginRight25": Object {
        "marginRight": 25,
      },
      "menuTriggerStyles": Object {
        "TriggerTouchableComponent": [Function],
        "triggerOuterWrapper": Object {
          "marginRight": 10,
        },
        "triggerText": Object {
          "color": "white",
        },
        "triggerTouchable": Object {
          "activeOpacity": 0.6,
          "underlayColor": "transparent",
        },
        "triggerWrapper": Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "justifyContent": "center",
        },
      },
      "moreNotificationButtonStyle": Object {
        "paddingBottom": 15,
        "paddingLeft": 15,
        "paddingRight": 15,
        "paddingTop": 25,
      },
      "moreNotificationTextStyle": Object {
        "color": "#007DBA",
        "fontSize": 13,
        "fontWeight": "bold",
      },
      "notificationHierarchy": Object {
        "color": "#444444",
        "fontWeight": "bold",
      },
      "notificationRow": Object {
        "flex": 1,
        "flexDirection": "row",
        "paddingBottom": 2,
        "paddingLeft": 2,
        "paddingRight": 2,
        "paddingTop": 2,
      },
      "notificationTextView": Object {
        "flexDirection": "column",
        "paddingLeft": 12,
        "paddingRight": 12,
      },
      "popoverProps": Object {
        "placement": "bottom",
        "preferredPlacement": "bottom",
      },
      "profileAvatar": Object {
        "borderRadius": 20,
        "height": 40,
        "marginRight": 10,
        "width": 40,
      },
      "profileMenuView": Object {
        "alignItems": "center",
        "borderBottomColor": "#C9C9C9",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "marginBottom": 2,
        "paddingBottom": 12,
      },
      "subText": Object {
        "color": "#939393",
        "fontSize": 11,
      },
      "timeStamp": Object {
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "width": 40,
      },
      "whiteText": Object {
        "color": "white",
      },
    }
  }
  title="Load Details"
  virtual={undefined}
/>
`;
